geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/Z3IdxStrategy.scala

```diff
@@ -117,31 +117,33 @@ class Z3IdxStrategy(val filter: QueryFilter) extends Strategy with Logging with
     // setup Z3 iterator
     val env = geometryToCover.getEnvelopeInternal
     val (lx, ly, ux, uy) = (env.getMinX, env.getMinY, env.getMaxX, env.getMaxY)
+
+    def qp(week: Int, lt: Long, ut: Long, contained: Boolean) =
+      queryPlanForPrefix(week, (lx, ux), (ly, uy), (lt, ut), z3table,
+        kvsToFeatures, iterators, colFamily, numThreads, contained)
+
     val epochWeekStart = Weeks.weeksBetween(Z3Table.EPOCH, interval.getStart)
     val epochWeekEnd = Weeks.weeksBetween(Z3Table.EPOCH, interval.getEnd)
     val weeks = scala.Range.inclusive(epochWeekStart.getWeeks, epochWeekEnd.getWeeks)
     val lt = Z3Table.secondsInCurrentWeek(interval.getStart, epochWeekStart)
     val ut = Z3Table.secondsInCurrentWeek(interval.getEnd, epochWeekEnd)
+
     if (weeks.length == 1) {
-      Seq(queryPlanForPrefix(weeks.head, lt ,ut, lx, ly, ux, uy,
-        z3table, kvsToFeatures, iterators, colFamily, numThreads, contained = false))
+      Seq(qp(weeks.head, lt, ut, contained = false))
     } else {
-      val oneWeekInSeconds = Weeks.ONE.toStandardSeconds.getSeconds
       val head +: xs :+ last = weeks.toList
-      val middleQPs = xs.map { w =>
-        queryPlanForPrefix(w, 0, oneWeekInSeconds, lx, ly, ux, uy,
-          z3table, kvsToFeatures, iterators, colFamily, numThreads, contained = true)
-      }
-      val startQP = queryPlanForPrefix(head, lt, oneWeekInSeconds, lx, ly, ux, uy,
-        z3table, kvsToFeatures, iterators, colFamily, numThreads, contained = false)
-      val endQP = queryPlanForPrefix(last, 0, ut, lx, ly, ux, uy,
-        z3table, kvsToFeatures, iterators, colFamily, numThreads, contained = false)
+      val oneWeekInSeconds = Weeks.ONE.toStandardSeconds.getSeconds
+      val startQP = qp(head, lt, oneWeekInSeconds, contained = false)
+      val endQP = qp(last, 0, ut, contained = false)
+      val middleQPs = xs.map(w => qp(w, 0, oneWeekInSeconds, contained = true))
```

Comment:
I know this isn't your code, but I'd consider renaming oneWeekInSeconds to something more helpful...